### ISSUE TITLE
undefined reference to `__errno_location' on FreeBSD 14.1

### ISSUE THREAD

**Issue by Martinfx** [ZathuraDbg/ZathuraDbg#94]
I used GCC 13 and i have still problem with undefined references

what is this https://github.com/ZathuraDbg/ZathuraDbg/blob/master/vendor/keystone/libkeystone.a ?
this is not going to be cross-platform

```
glfw  -lGL  -ldl  -lstdc++  -lunicorn  -L/wrkdirs/usr/ports/devel/zathuradbg/work/ZathuraDbg-0.1.0-beta/src/../vendor/keystone  -l:libkeystone.a  -L/wrkdirs/usr/ports/devel/zathuradbg/work/ZathuraDbg-0.1.0-beta/src/../vendor/capstone  -l:libcapstone.a  -pthread  -std=c++23 && :
/usr/local/bin/ld: /wrkdirs/usr/ports/devel/zathuradbg/work/ZathuraDbg-0.1.0-beta/src/../vendor/keystone/libkeystone.a(MemoryBuffer.cpp.o): in function `getMemoryBufferForStream(int, llvm_ks::Twine const&)':
MemoryBuffer.cpp:(.text._ZL24getMemoryBufferForStreamiRKN7llvm_ks5TwineE+0xd1): undefined reference to `__errno_location'
/usr/local/bin/ld: /wrkdirs/usr/ports/devel/zathuradbg/work/ZathuraDbg-0.1.0-beta/src/../vendor/keystone/libkeystone.a(MemoryBuffer.cpp.o): in function `getOpenFileImpl(int, llvm_ks::Twine const&, unsigned long, unsigned long, long, bool, bool)':
MemoryBuffer.cpp:(.text._ZL15getOpenFileImpliRKN7llvm_ks5TwineEmmlbb+0x2e1): undefined reference to `__errno_location'
/usr/local/bin/ld: /wrkdirs/usr/ports/devel/zathuradbg/work/ZathuraDbg-0.1.0-beta/src/../vendor/keystone/libkeystone.a(Path.cpp.o): in function `llvm_ks::sys::fs::create_directory(llvm_ks::Twine const&, bool, llvm_ks::sys::fs::perms)':
Path.cpp:(.text._ZN7llvm_ks3sys2fs16create_directoryERKNS_5TwineEbNS1_5permsE+0xa1): undefined reference to `__errno_location'
/usr/local/bin/ld: /wrkdirs/usr/ports/devel/zathuradbg/work/ZathuraDbg-0.1.0-beta/src/../vendor/keystone/libkeystone.a(Path.cpp.o): in function `llvm_ks::sys::fs::create_link(llvm_ks::Twine const&, llvm_ks::Twine const&)':
Path.cpp:(.text._ZN7llvm_ks3sys2fs11create_linkERKNS_5TwineES4_+0xf9): undefined reference to `__errno_location'
/usr/local/bin/ld: /wrkdirs/usr/ports/devel/zathuradbg/work/ZathuraDbg-0.1.0-beta/src/../vendor/keystone/libkeystone.a(Path.cpp.o): in function `llvm_ks::sys::fs::remove(llvm_ks::Twine const&, bool)':
Path.cpp:(.text._ZN7llvm_ks3sys2fs6removeERKNS_5TwineEb+0xe1): undefined reference to `__errno_location'
/usr/local/bin/ld: /wrkdirs/usr/ports/devel/zathuradbg/work/ZathuraDbg-0.1.0-beta/src/../vendor/keystone/libkeystone.a(Path.cpp.o):Path.cpp:(.text._ZN7llvm_ks3sys2fs6renameERKNS_5TwineES4_+0xf9): more undefined references to `__errno_location' follow
/usr/local/bin/ld: /wrkdirs/usr/ports/devel/zathuradbg/work/ZathuraDbg-0.1.0-beta/src/../vendor/keystone/libkeystone.a(Path.cpp.o): in function `llvm_ks::sys::fs::getMainExecutable[abi:cxx11](char const*, void*)':
Path.cpp:(.text._ZN7llvm_ks3sys2fs17getMainExecutableB5cxx11EPKcPv+0x1e6): undefined reference to `__snprintf_chk'
/usr/local/bin/ld: Path.cpp:(.text._ZN7llvm_ks3sys2fs17getMainExecutableB5cxx11EPKcPv+0x1f6): undefined reference to `__realpath_chk'
/usr/local/bin/ld: Path.cpp:(.text._ZN7llvm_ks3sys2fs17getMainExecutableB5cxx11EPKcPv+0x368): undefined reference to `__snprintf_chk'
/usr/local/bin/ld: Path.cpp:(.text._ZN7llvm_ks3sys2fs17getMainExecutableB5cxx11EPKcPv+0x380): undefined reference to `__realpath_chk'
/usr/local/bin/ld: /wrkdirs/usr/ports/devel/zathuradbg/work/ZathuraDbg-0.1.0-beta/src/../vendor/keystone/libkeystone.a(Path.cpp.o): in function `llvm_ks::sys::fs::status(llvm_ks::Twine const&, llvm_ks::sys::fs::file_status&)':
Path.cpp:(.text._ZN7llvm_ks3sys2fs6statusERKNS_5TwineERNS1_11file_statusE+0x8a): undefined reference to `__errno_location'
/usr/local/bin/ld: /wrkdirs/usr/ports/devel/zathuradbg/work/ZathuraDbg-0.1.0-beta/src/../vendor/keystone/libkeystone.a(Path.cpp.o): in function `llvm_ks::sys::fs::current_path(llvm_ks::SmallVectorImpl<char>&)':
Path.cpp:(.text._ZN7llvm_ks3sys2fs12current_pathERNS_15SmallVectorImplIcEE+0x10e): undefined reference to `__errno_location'
/usr/local/bin/ld: /wrkdirs/usr/ports/devel/zathuradbg/work/ZathuraDbg-0.1.0-beta/src/../vendor/keystone/libkeystone.a(Path.cpp.o): in function `llvm_ks::sys::fs::status(int, llvm_ks::sys::fs::file_status&)':
Path.cpp:(.text._ZN7llvm_ks3sys2fs6statusEiRNS1_11file_statusE+0x4b): undefined reference to `__errno_location'
/usr/local/bin/ld: /wrkdirs/usr/ports/devel/zathuradbg/work/ZathuraDbg-0.1.0-beta/src/../vendor/keystone/libkeystone.a(Path.cpp.o): in function `llvm_ks::sys::fs::mapped_file_region::init(int, unsigned long, llvm_ks::sys::fs::mapped_file_region::mapmode)':
Path.cpp:(.text._ZN7llvm_ks3sys2fs18mapped_file_region4initEimNS2_7mapmodeE+0x61): undefined reference to `__errno_location'
/usr/local/bin/ld: /wrkdirs/usr/ports/devel/zathuradbg/work/ZathuraDbg-0.1.0-beta/src/../vendor/keystone/libkeystone.a(Path.cpp.o): in function `llvm_ks::sys::fs::mapped_file_region::mapped_file_region(int, llvm_ks::sys::fs::mapped_file_region::mapmode, unsigned long, unsigned long, std::error_code&)':
Path.cpp:(.text._ZN7llvm_ks3sys2fs18mapped_file_regionC2EiNS2_7mapmodeEmmRSt10error_code+0x79): undefined reference to `__errno_location'
/usr/local/bin/ld: /wrkdirs/usr/ports/devel/zathuradbg/work/ZathuraDbg-0.1.0-beta/src/../vendor/keystone/libkeystone.a(Path.cpp.o):Path.cpp:(.text._ZN7llvm_ks3sys2fs6detail28directory_iterator_incrementERNS2_12DirIterStateE+0x20): more undefined references to `__errno_location' follow
/usr/local/bin/ld: /wrkdirs/usr/ports/devel/zathuradbg/work/ZathuraDbg-0.1.0-beta/src/../vendor/keystone/libkeystone.a(raw_ostream.cpp.o): in function `llvm_ks::format_object<double>::snprint(char*, unsigned int) const':
raw_ostream.cpp:(.text._ZNK7llvm_ks13format_objectIJdEE7snprintEPcj[_ZNK7llvm_ks13format_objectIJdEE7snprintEPcj]+0x27): undefined reference to `__snprintf_chk'
/usr/local/bin/ld: /wrkdirs/usr/ports/devel/zathuradbg/work/ZathuraDbg-0.1.0-beta/src/../vendor/keystone/libkeystone.a(raw_ostream.cpp.o): in function `llvm_ks::raw_fd_ostream::write_impl(char const*, unsigned long)':
raw_ostream.cpp:(.text._ZN7llvm_ks14raw_fd_ostream10write_implEPKcm+0x41): undefined reference to `__errno_location'
/usr/local/bin/ld: /wrkdirs/usr/ports/devel/zathuradbg/work/ZathuraDbg-0.1.0-beta/src/../vendor/keystone/libkeystone.a(raw_ostream.cpp.o): in function `llvm_ks::raw_fd_ostream::~raw_fd_ostream()':
raw_ostream.cpp:(.text._ZN7llvm_ks14raw_fd_ostreamD0Ev+0x60): undefined reference to `__errno_location'
/usr/local/bin/ld: /wrkdirs/usr/ports/devel/zathuradbg/work/ZathuraDbg-0.1.0-beta/src/../vendor/keystone/libkeystone.a(raw_ostream.cpp.o): in function `llvm_ks::raw_fd_ostream::~raw_fd_ostream()':
raw_ostream.cpp:(.text._ZN7llvm_ks14raw_fd_ostreamD2Ev+0x60): undefined reference to `__errno_location'
/usr/local/bin/ld: /wrkdirs/usr/ports/devel/zathuradbg/work/ZathuraDbg-0.1.0-beta/src/../vendor/keystone/libkeystone.a(raw_ostream.cpp.o): in function `llvm_ks::raw_fd_ostream::pwrite_impl(char const*, unsigned long, unsigned long)':
raw_ostream.cpp:(.text._ZN7llvm_ks14raw_fd_ostream11pwrite_implEPKcmm+0x141): undefined reference to `__errno_location'
/usr/local/bin/ld: raw_ostream.cpp:(.text._ZN7llvm_ks14raw_fd_ostream11pwrite_implEPKcmm+0x169): undefined reference to `__errno_location'
/usr/local/bin/ld: /wrkdirs/usr/ports/devel/zathuradbg/work/ZathuraDbg-0.1.0-beta/src/../vendor/keystone/libkeystone.a(raw_ostream.cpp.o): in function `llvm_ks::raw_ostream::operator<<(double)':
raw_ostream.cpp:(.text._ZN7llvm_ks11raw_ostreamlsEd+0x71): undefined reference to `__snprintf_chk'
/usr/local/bin/ld: raw_ostream.cpp:(.text._ZN7llvm_ks11raw_ostreamlsEd+0x12a): undefined reference to `__snprintf_chk'
/usr/local/bin/ld: /wrkdirs/usr/ports/devel/zathuradbg/work/ZathuraDbg-0.1.0-beta/src/../vendor/keystone/libkeystone.a(raw_ostream.cpp.o): in function `llvm_ks::raw_ostream::operator<<(llvm_ks::format_object_base const&)':
raw_ostream.cpp:(.text._ZN7llvm_ks11raw_ostreamlsERKNS_18format_object_baseE+0x8a): undefined reference to `__snprintf_chk'
/usr/local/bin/ld: raw_ostream.cpp:(.text._ZN7llvm_ks11raw_ostreamlsERKNS_18format_object_baseE+0x131): undefined reference to `__snprintf_chk'
/usr/local/bin/ld: /wrkdirs/usr/ports/devel/zathuradbg/work/ZathuraDbg-0.1.0-beta/src/../vendor/keystone/libkeystone.a(raw_ostream.cpp.o): in function `llvm_ks::raw_fd_ostream::close()':
raw_ostream.cpp:(.text._ZN7llvm_ks14raw_fd_ostream5closeEv+0x69): undefined reference to `__errno_location'
/usr/local/bin/ld: /wrkdirs/usr/ports/devel/zathuradbg/work/ZathuraDbg-0.1.0-beta/src/../vendor/keystone/libkeystone.a(raw_ostream.cpp.o): in function `llvm_ks::raw_fd_ostream::seek(unsigned long)':
raw_ostream.cpp:(.text._ZN7llvm_ks14raw_fd_ostream4seekEm+0x91): undefined reference to `__errno_location'
/usr/local/bin/ld: /wrkdirs/usr/ports/devel/zathuradbg/work/ZathuraDbg-0.1.0-beta/src/../vendor/keystone/libkeystone.a(SubtargetFeature.cpp.o): in function `llvm_ks::format_object<unsigned int, char const*, char const*>::snprint(char*, unsigned int) const':
SubtargetFeature.cpp:(.text._ZNK7llvm_ks13format_objectIJjPKcS2_EE7snprintEPcj[_ZNK7llvm_ks13format_objectIJjPKcS2_EE7snprintEPcj]+0x2d): undefined reference to `__snprintf_chk'
/usr/local/bin/ld: /wrkdirs/usr/ports/devel/zathuradbg/work/ZathuraDbg-0.1.0-beta/src/../vendor/capstone/libcapstone.a(ARMInstPrinter.o): in function `ARMCondCodeToString.part.0':
ARMInstPrinter.c:(.text+0xdf): undefined reference to `__assert_fail'
/usr/local/bin/ld: /wrkdirs/usr/ports/devel/zathuradbg/work/ZathuraDbg-0.1.0-beta/src/../vendor/capstone/libcapstone.a(ARMInstPrinter.o): in function `ARM_AM_getShiftOpcStr.part.0':
ARMInstPrinter.c:(.text+0x10f): undefined reference to `__assert_fail'
/usr/local/bin/ld: /wrkdirs/usr/ports/devel/zathuradbg/work/ZathuraDbg-0.1.0-beta/src/../vendor/capstone/libcapstone.a(ARMInstPrinter.o): in function `printRegName':
ARMInstPrinter.c:(.text+0x46a): undefined reference to `__assert_fail'
/usr/local/bin/ld: ARMInstPrinter.c:(.text+0x489): undefined reference to `__assert_fail'
/usr/local/bin/ld: /wrkdirs/usr/ports/devel/zathuradbg/work/ZathuraDbg-0.1.0-beta/src/../vendor/capstone/libcapstone.a(ARMInstPrinter.o): in function `printOperand':
ARMInstPrinter.c:(.text+0x55f): undefined reference to `__assert_fail'
/usr/local/bin/ld: /wrkdirs/usr/ports/devel/zathuradbg/work/ZathuraDbg-0.1.0-beta/src/../vendor/capstone/libcapstone.a(ARMInstPrinter.o):ARMInstPrinter.c:(.text+0x66c): more undefined references to `__assert_fail' follow
/usr/local/bin/ld: /wrkdirs/usr/ports/devel/zathuradbg/work/ZathuraDbg-0.1.0-beta/src/../vendor/keystone/libkeystone.a(regcomp.c.o): in function `othercase':
regcomp.c:(.text.othercase+0xc): undefined reference to `__ctype_b_loc'
/usr/local/bin/ld: regcomp.c:(.text.othercase+0x24): undefined reference to `__ctype_toupper_loc'
/usr/local/bin/ld: regcomp.c:(.text.othercase+0x41): undefined reference to `__ctype_tolower_loc'
/usr/local/bin/ld: /wrkdirs/usr/ports/devel/zathuradbg/work/ZathuraDbg-0.1.0-beta/src/../vendor/keystone/libkeystone.a(regcomp.c.o): in function `p_bracket':
regcomp.c:(.text.p_bracket+0x426): undefined reference to `__ctype_b_loc'
/usr/local/bin/ld: regcomp.c:(.text.p_bracket+0x7c5): undefined reference to `__ctype_b_loc'
/usr/local/bin/ld: /wrkdirs/usr/ports/devel/zathuradbg/work/ZathuraDbg-0.1.0-beta/src/../vendor/keystone/libkeystone.a(regcomp.c.o): in function `ordinary':
regcomp.c:(.text.ordinary+0x32): undefined reference to `__ctype_b_loc'
/usr/local/bin/ld: /wrkdirs/usr/ports/devel/zathuradbg/work/ZathuraDbg-0.1.0-beta/src/../vendor/keystone/libkeystone.a(regcomp.c.o): in function `p_bre':
regcomp.c:(.text.p_bre+0x5d8): undefined reference to `__ctype_b_loc'
/usr/local/bin/ld: /wrkdirs/usr/ports/devel/zathuradbg/work/ZathuraDbg-0.1.0-beta/src/../vendor/keystone/libkeystone.a(regcomp.c.o): in function `p_ere':
regcomp.c:(.text.p_ere+0xeb): undefined reference to `__ctype_b_loc'
/usr/local/bin/ld: /wrkdirs/usr/ports/devel/zathuradbg/work/ZathuraDbg-0.1.0-beta/src/../vendor/keystone/libkeystone.a(regcomp.c.o):regcomp.c:(.text.p_ere+0x171): more undefined references to `__ctype_b_loc' follow
/usr/local/bin/ld: /wrkdirs/usr/ports/devel/zathuradbg/work/ZathuraDbg-0.1.0-beta/src/../vendor/keystone/libkeystone.a(regerror.c.o): in function `llvm_regerror':
regerror.c:(.text.llvm_regerror+0x13e): undefined reference to `__snprintf_chk'
/usr/local/bin/ld: /wrkdirs/usr/ports/devel/zathuradbg/work/ZathuraDbg-0.1.0-beta/src/../vendor/keystone/libkeystone.a(regexec.c.o): in function `sslow':
regexec.c:(.text.sslow+0xf8): undefined reference to `__ctype_b_loc'
/usr/local/bin/ld: regexec.c:(.text.sslow+0x21d): undefined reference to `__ctype_b_loc'
/usr/local/bin/ld: regexec.c:(.text.sslow+0x2ce): undefined reference to `__ctype_b_loc'
/usr/local/bin/ld: regexec.c:(.text.sslow+0x34b): undefined reference to `__ctype_b_loc'
/usr/local/bin/ld: regexec.c:(.text.sslow+0x383): undefined reference to `__ctype_b_loc'
/usr/local/bin/ld: /wrkdirs/usr/ports/devel/zathuradbg/work/ZathuraDbg-0.1.0-beta/src/../vendor/keystone/libkeystone.a(regexec.c.o):regexec.c:(.text.sslow+0x57b): more undefined references to `__ctype_b_loc' follow
/usr/local/bin/ld: /wrkdirs/usr/ports/devel/zathuradbg/work/ZathuraDbg-0.1.0-beta/src/../vendor/capstone/libcapstone.a(utils.o): in function `readBytes32':
utils.c:(.text+0x21f): undefined reference to `__assert_fail'
/usr/local/bin/ld: /wrkdirs/usr/ports/devel/zathuradbg/work/ZathuraDbg-0.1.0-beta/src/../vendor/capstone/libcapstone.a(utils.o): in function `readBytes16':
utils.c:(.text+0x271): undefined reference to `__assert_fail'
/usr/local/bin/ld: /wrkdirs/usr/ports/devel/zathuradbg/work/ZathuraDbg-0.1.0-beta/src/../vendor/capstone/libcapstone.a(utils.o): in function `append_to_str_lower':
utils.c:(.text+0x2b1): undefined reference to `__ctype_tolower_loc'
/usr/local/bin/ld: utils.c:(.text+0x309): undefined reference to `__assert_fail'
/usr/local/bin/ld: /wrkdirs/usr/ports/devel/zathuradbg/work/ZathuraDbg-0.1.0-beta/src/../vendor/capstone/libcapstone.a(SStream.o): in function `SStream_Init':
SStream.c:(.text+0x36): undefined reference to `__assert_fail'
/usr/local/bin/ld: /wrkdirs/usr/ports/devel/zathuradbg/work/ZathuraDbg-0.1.0-beta/src/../vendor/capstone/libcapstone.a(SStream.o): in function `SStream_Open':
SStream.c:(.text+0x69): undefined reference to `__assert_fail'
/usr/local/bin/ld: /wrkdirs/usr/ports/devel/zathuradbg/work/ZathuraDbg-0.1.0-beta/src/../vendor/capstone/libcapstone.a(SStream.o): in function `SStream_Close':
SStream.c:(.text+0x99): undefined reference to `__assert_fail'
/usr/local/bin/ld: /wrkdirs/usr/ports/devel/zathuradbg/work/ZathuraDbg-0.1.0-beta/src/../vendor/capstone/libcapstone.a(MCInst.o): in function `MCInst_insert0':
MCInst.c:(.text+0x136): undefined reference to `__assert_fail'
/usr/local/bin/ld: /wrkdirs/usr/ports/devel/zathuradbg/work/ZathuraDbg-0.1.0-beta/src/../vendor/capstone/libcapstone.a(MCInst.o):MCInst.c:(.text+0x1b0): more undefined references to `__assert_fail' follow

```

**Comment by HACKE-RC** [comment]
I believe I know the cause of this issue. I'll probably look at it in a few hours.

**Comment by HACKE-RC** [comment]
Seems like I forgot to update the CMakeFile.txt to remove the static linking of the keystone static library. I will fix this, thanks for reporting it!

**Comment by HACKE-RC** [comment]
I assume this fixed it: https://github.com/ZathuraDbg/ZathuraDbg/commit/b2de6674fa130b158b0d448b42fe0423bf13a6a6

**Comment by Martinfx** [comment]
> I assume this fixed it: [b2de667](https://github.com/ZathuraDbg/ZathuraDbg/commit/b2de6674fa130b158b0d448b42fe0423bf13a6a6)

thank you.
